app/scripts/modules/amazon/src/instance/details/InstanceDns.tsx

```diff
@@ -0,0 +1,66 @@
+import * as React from 'react';
+import { CollapsibleSection, LabeledValue, LabeledValueList, LinkWithClipboard } from '@spinnaker/core';
+
+interface IIpv6Address {
+  ip: string;
+  url: string;
+}
+
+export interface IInstanceDnsProps {
+  instancePort: string;
+  ipv6Addresses: IIpv6Address[];
+  permanentIps: string[];
+  privateDnsName: string;
+  privateIpAddress: string;
+  publicDnsName: string;
+  publicIpAddress: string;
+}
+
+export const InstanceDns = ({
+  instancePort,
+  ipv6Addresses,
+  permanentIps,
+  privateDnsName,
+  privateIpAddress,
+  publicDnsName,
+  publicIpAddress,
+}: IInstanceDnsProps) => (
+  <CollapsibleSection heading="DNS" defaultExpanded={true}>
+    <LabeledValueList className="horizontal-when-filters-collapsed">
+      {privateDnsName && (
+        <LabeledValue
+          label="Private DNS Name"
+          value={<LinkWithClipboard text={privateDnsName} url={`http://${privateDnsName}:${instancePort}`} />}
+        />
+      )}
+      {publicDnsName && (
+        <LabeledValue
+          label="Public DNS Name"
+          value={<LinkWithClipboard text={publicDnsName} url={`http://${publicDnsName}:${instancePort}`} />}
+        />
+      )}
+      {privateIpAddress && (
+        <LabeledValue
+          label="Private IP Address"
+          value={<LinkWithClipboard text={privateIpAddress} url={`http://${privateIpAddress}:${instancePort}`} />}
+        />
+      )}
+      {permanentIps && <dt>Permanent IP Address</dt>}
+      <dd>
+        {permanentIps &&
+          permanentIps.map((ip) => <LinkWithClipboard key={ip} text={ip} url={`http://${ip}:${instancePort}`} />)}
+      </dd>
```

Comment:
Wondering if there was a reason you didn't use LabeledValue here?

```suggestion
      {permanentIps?.length && (
        <LabeledValue
          label="Permanent IP Address"
          value={permanentIps.map((ip) => <LinkWithClipboard key={ip} text={ip} url={`http://${ip}:${instancePort}`} />)}}
        />
      )}
```